### ISSUE TITLE
Can't disable stacks that have descendents

### ISSUE THREAD

**Issue by ejholmes** [remind101/stacker#383]
This is in 0.8.6, so maybe fixed in 1.0 already?

You can disable a stack by setting `enabled: false` within the stack configuration, however, if you have a disabled stack, that depends on another disabled stack, you'll get an error.

For example, you can't actually do this:

```yaml
stacks:
  - name: db
     enabled: false
  - name: app
     enabled: false
     variables:
       DB: ${output db::Address}
```

Because, you'll get an error like:

```
  File "build/bdist.linux-x86_64/egg/stacker/commands/stacker/build.py", line 52, in run
  File "build/bdist.linux-x86_64/egg/stacker/actions/base.py", line 122, in execute
  File "build/bdist.linux-x86_64/egg/stacker/actions/build.py", line 311, in run
  File "build/bdist.linux-x86_64/egg/stacker/plan.py", line 251, in execute
  File "build/bdist.linux-x86_64/egg/stacker/plan.py", line 219, in _single_run
  File "build/bdist.linux-x86_64/egg/stacker/plan.py", line 72, in run
  File "build/bdist.linux-x86_64/egg/stacker/actions/build.py", line 209, in _launch_stack
  File "build/bdist.linux-x86_64/egg/stacker/stack.py", line 180, in resolve_variables
  File "build/bdist.linux-x86_64/egg/stacker/variables.py", line 74, in resolve_variables
  File "build/bdist.linux-x86_64/egg/stacker/variables.py", line 134, in resolve
  File "build/bdist.linux-x86_64/egg/stacker/lookups/registry.py", line 50, in resolve_lookups
  File "build/bdist.linux-x86_64/egg/stacker/lookups/handlers/output.py", line 35, in handler
  File "build/bdist.linux-x86_64/egg/stacker/providers/base.py", line 33, in get_output
  File "build/bdist.linux-x86_64/egg/stacker/providers/aws/default.py", line 247, in get_outputs
  File "build/bdist.linux-x86_64/egg/stacker/providers/aws/default.py", line 115, in get_stack
stacker.exceptions.StackDoesNotExist: Stack: "db" does not exist in outputs
```

**Comment by phobologic** [comment]
Interesting - yeah, we just need to disable lookup resolution on disabled stacks.  Makes sense anyways.

**Comment by phobologic** [comment]
I'm really confused as to how it's even getting to line 209 (https://github.com/remind101/stacker/blob/0.8.6/stacker/actions/build.py#L209) if the stack is disabled, since this should skip the stack altogether: https://github.com/remind101/stacker/blob/0.8.6/stacker/actions/build.py#L186

Very strange.  Was set to say this was fixed in 1.0, since that's the logic I saw there, but if this is happening then something is obviously broken.

**Comment by phobologic** [comment]
Hmm - haven't been able to reproduce this with a simple config, and @ejholmes thinks it may have been something else that was breaking, but in an expected way.  The bigger issue here is that in 0.8.6 we never gave a good idea of where something was breaking in the config - this has been worked on quite a bit in 1.X, so hopefully this should be easier once we're all moved.
